node-services/cards-external-diffusion/src/domain/server-side/opfabInterface.ts

```diff
@@ -0,0 +1,315 @@
+            }
+        } catch (e) {
+            this.logger.warn('Impossible to get users', e);
+            return new GetResponse([], false);
+        }
+       
+    }
+
+    public async getUsersConnected(): Promise<GetResponse> {
+        try {
+            await this.getToken();
+            const response = await this.sendUsersConnectedRequest();
+            const users = new Array();
+            if (response?.data) {
+                response.data.forEach((user: any) => {
+                    users.push(user.login);
+                });
+                return new GetResponse(users, true);
+            }
+            else {
+                this.logger.warn("No data in HTTP response")
+                return new GetResponse(null, false);
+            }
+        } catch (e) {
+            this.logger.warn('Impossible to get connected users', e);
+            return new GetResponse(null, false);
+        }
+       
+    }
+
+    public async getUserSettings(login: string): Promise<GetResponse> {
+        try {
+            if (this.userSettings.has(login))
+                return new GetResponse(this.userSettings.get(login), true);
+
+            await this.getToken();
+            const response = await this.sendUserSettingRequest(login);
+            if (response?.data) {
+                this.userSettings.set(login, response.data);
+                return new GetResponse(response.data, true);
+            }
+            else {
+                this.logger.warn("No data in HTTP response")
+                return new GetResponse(null, false);
+            }
+        } catch (error: any) {
+            if (error.response.status != 404) {
+                this.logger.warn('Impossible to get user settings', error);
+            }
+
+            return new GetResponse(null, false);
+        }
+    }
+
+    public async getUserPerimeters(login: string): Promise<GetResponse> {
+        try {
+            if (this.userPerimeters.has(login))
+                return new GetResponse(this.userPerimeters.get(login), true);
+
+            await this.getToken();
+            const response = await this.sendUserPerimetersRequest(login);
+            if (response?.data) {
+                this.userPerimeters.set(login, response.data);
+                return new GetResponse(response.data, true);
+            }
+            else {
+                this.logger.warn("No data in HTTP response")
+                return new GetResponse(null, false);
+            }
+        } catch (error: any) {
+            this.logger.warn('Impossible to get user perimeters', error);
+            return new GetResponse(null, false);
+        }
+    }
+
+
+    public loadUsersData() {
+        this.fetchAllUsers();
+        this.userPerimeters.clear();
+        this.userSettings.clear();
+    }
+
+    public clearUserCache(login: string) {
+        this.userSettings.delete(login);
+        this.userPerimeters.delete(login);
+        this.fetchUser(login);
```

Comment:
Maybe we do not need to fetch here , it will be fetch after if missing

This avoid pb when a lot of updates came from rabbit 

To discuss

Reply:
In fact 'fecthUser()' it will not be called if the user is missing from cache (normally the control takes the users list from the cache),  so it has to be called when removing a user from the cache.